### ISSUE TITLE
BUG: No keyword arguments passed to numpy.interp

### ISSUE THREAD

**Issue by kmuehlbauer** [pandas-dev/pandas#55144]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np
df = pd.DataFrame([np.nan, 0, 1, 2, np.nan, 3, 4, 5, np.nan])
out = df.interpolate(method="linear", left=10, right=20)
print(out)
     0
0  NaN
1  0.0
2  1.0
3  2.0
4  2.5
5  3.0
6  4.0
7  5.0
8  5.0
```

### Issue Description

This bug is connected to #31949, at least with respect to `numpy.interp`.

The ideal user solution would be to translate `fill_value` here to `numpy.interp` `left`/`right` kwargs.

### Expected Behavior

```python
     0
0  10.0
1   0.0
2   1.0
3   2.0
4   2.5
5   3.0
6   4.0
7   5.0
8  20.0
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit              : ba1cccd19da778f0c3a7d6a885685da16a072870
python              : 3.11.0.final.0
python-bits         : 64
OS                  : Linux
OS-release          : 5.14.21-150500.55.19-default
Version             : #1 SMP PREEMPT_DYNAMIC Tue Aug 8 22:15:01 UTC 2023 (9908c29)
machine             : x86_64
processor           : x86_64
byteorder           : little
LC_ALL              : None
LANG                : de_DE.UTF-8
LOCALE              : de_DE.UTF-8

pandas              : 2.1.0
numpy               : 1.24.2
pytz                : 2022.7.1
dateutil            : 2.8.2
setuptools          : 67.6.0
pip                 : 23.0.1
Cython              : None
pytest              : 7.2.2
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : None
lxml.etree          : None
html5lib            : None
pymysql             : None
psycopg2            : None
jinja2              : 3.1.2
IPython             : 8.11.0
pandas_datareader   : None
bs4                 : 4.12.0
bottleneck          : 1.3.7
dataframe-api-compat: None
fastparquet         : None
fsspec              : 2023.3.0
gcsfs               : None
matplotlib          : 3.7.1
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : None
pandas_gbq          : None
pyarrow             : 11.0.0
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : 1.10.1
sqlalchemy          : None
tables              : None
tabulate            : None
xarray              : 2023.8.1.dev8+g1809b913.d20230831
xlrd                : None
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None

</details>

**Comment by kmuehlbauer** [comment]
Anyone?

**Comment by rhshadrach** [comment]
With all the complexities of pandas' interpolate (namely, `limit_area` and `limit_direction`), I do not think we can just pass these down to NumPy. It seems to me `kwargs` were only ever intended to be used on SciPy interpolators. I think a PR to make this clear in the docs would be welcome.

Perhaps pandas could support extrapolation better here, including `left` / `right`. It seems to me that #31949 is the proper issue for that.

**Comment by kmuehlbauer** [comment]
Thanks @rhshadrach for taking a look here. I actually came from #31949, see https://github.com/pandas-dev/pandas/issues/31949#issuecomment-1721003085. So yes, would be great if this could be worked out over in #31949. That issue is approaching 6 years of existence in near time without gaining interest by pandas devs so far.

**Comment by rhshadrach** [comment]
@kmuehlbauer pandas is almost entirely a volunteer developed project, I would guess it's quite likely it will go another 6 years unless someone with interest in fixing this puts together a plan and a PR.

**Comment by kmuehlbauer** [comment]
Yes, another six years might indeed be the case. As my concern is only with `numpy.interp`, I'm still happy to help to resolve that particular issue. 

@rhshadrach maybe you can comment a bit more, why `numpy.interp` should not receive the needed kwargs (`left`,`right`)? At least there could be `fill_value` distributed to these two kwargs, to get at least a minimal solution. Thanks!

**Comment by rhshadrach** [comment]
> why `numpy.interp` should not receive the needed kwargs (`left`,`right`)?

My comment was saying that I believe you cannot _just_ pass to NumPy. Trying that out, I am seeing

```python
df = pd.DataFrame([np.nan, 0, np.nan, 1, np.nan])

print(df.interpolate(method="linear", left=10, right=20))
#       0
# 0   NaN
# 1   0.0
# 2   0.5
# 3   1.0
# 4  20.0

print(df.interpolate(method="linear", left=10, right=20, limit_area="inside"))
#      0
# 0  NaN
# 1  0.0
# 2  0.5
# 3  1.0
# 4  NaN

print(df.interpolate(method="linear", left=10, right=20, limit_direction="backward"))
#       0
# 0  10.0
# 1   0.0
# 2   0.5
# 3   1.0
# 4   NaN
```

which I think are incorrect results.

Also I imagine we would want to raise if `period` (or the first three named arguments) is passed in kwargs. Finally, we would need to ensure this is implemented properly across the API for consistency (e.g. `Resampler.interpolate`).

**Comment by kmuehlbauer** [comment]
Thanks for fixing this in the docstring. At least this resolves any possible misunderstanding.
